core/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/scheduling/ScheduleManagerImpl.java

```diff
@@ -151,11 +167,16 @@ private void tryRunNext()
         {
             for (ScheduledJob next : myQueue)
             {
-                if (validate(next) && tryRunTasks(next))
+                if (validate(next))
                 {
-                    break;
+                    currentExecutingJob.set(next); // Set the currently executing job
```

Comment:
If comment is necessary, it should probably state why it set rather than it is set.

Reply:
Done